persistence-api/src/main/java/io/stargate/db/tracing/QueryTracingFetcher.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.tracing;
+
+import io.stargate.db.Parameters;
+import io.stargate.db.Persistence;
+import io.stargate.db.Result;
+import io.stargate.db.Result.Rows;
+import io.stargate.db.SimpleStatement;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+
+public class QueryTracingFetcher {
+
+  private final UUID tracingId;
+  private final Persistence.Connection connection;
+  private final CompletableFuture<Rows> resultFuture = new CompletableFuture<>();
+  private static final ConsistencyLevel TRACE_CONSISTENCY = ConsistencyLevel.ONE;
+  private static final int REQUEST_TRACE_ATTEMPTS = 5;
+  private static final Duration TRACE_INTERVAL = Duration.ofMillis(3);
+  private final Parameters parameters;
+  private final ScheduledExecutorService executorService;
+  private final ByteBuffer tracingIdBytes;
+
+  public QueryTracingFetcher(UUID tracingId, Persistence.Connection connection) {
+    this.tracingId = tracingId;
+    this.tracingIdBytes = decompose(tracingId);
+    this.connection = connection;
+    this.parameters = createTracingQueryParameters();
+    this.executorService = Executors.newSingleThreadScheduledExecutor();
```

Comment:
This is not ideal. 
Right now it will create a dedicated executor per tracing invocation. We may consider creating one executor with a higher number of threads, wdyt?

Reply:
Agreed, a shared executor would be preferred. I'm wondering if there's an existing one we can use?

Reply:
I've changed the logic to use one static `ScheduledExecutorService` here. It is created with the same config, as the java-driver scheduler for this component. Since the defaults are proven for drivers, it should meet our needs here as well.